src/blocks/scratch3_procedures.js

```diff
@@ -4,6 +4,8 @@ function Scratch3ProcedureBlocks(runtime) {
      * @type {Runtime}
      */
     this.runtime = runtime;
+    this.REPORT = {};
+    this.currentCall = null;
```

Comment:
We can't store data on the `Scratch3ProcedureBlocks` object, as I previously mentioned. Any data that's relevant for procedure calls needs to be associated with the thread.